### PR TITLE
fix crash when waypoint is not network address type

### LINKED ISSUE #815: Panic with istio 1.24-alpha.96593644e05baa5c1b6aaab3647b2d9335a23942
```
k get pod   -n kmesh-system 
NAME          READY   STATUS             RESTARTS      AGE
kmesh-5dcnv   0/1     CrashLoopBackOff   3 (24s ago)   2m28s
```

time="2024-09-05T04:20:28Z" level=warning msg="exceed the max port count,current only support maximum of 10 ports" subsys=workload_controller
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x556bffcab31f]

goroutine 101 [running]:
kmesh.net/kmesh/pkg/controller/workload.(*Processor).handleService(0xc0000c91a0, 0xc000c08210)
	/kmesh/pkg/controller/workload/workload_processor.go:506 +0xdf
kmesh.net/kmesh/pkg/controller/workload.(*Processor).handleAddressTypeResponse(0xc0000c91a0, 0xc0001124d0)
	/kmesh/pkg/controller/workload/workload_processor.go:573 +0x4d2

image 

```
      image: ghcr.io/kmesh-net/kmesh:latest
      imageID: ghcr.io/kmesh-net/kmesh@sha256:9ec2befa7f60f336a5c667ab35234a8d061af92b20ff3edd65e67be3e63613b1

```

### PR DESCRIPTION
**Pull request by YaoZengzeng**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #815

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/controller/workload/workload_processor.go [hunk 1]
```diff
@@ -501,7 +501,7 @@ func (p *Processor) handleService(service *workloadapi.Service) error {
 	}
 
 	// Preprocess service, remove the waypoint from waypoint service, otherwise it will fall into a loop in bpf
-	if service.Waypoint != nil {
+	if service.Waypoint != nil && service.GetWaypoint().GetAddress() != nil && len(service.Addresses) != 0 {
```
Comment (hzxuzhonghu) [inline pkg/controller/workload/workload_processor.go:504]:
```suggestion
	if service.Waypoint != nil && len(service.GetWaypoint().GetAddress()) > 0 {
```

Comment (YaoZengzeng) [reply pkg/controller/workload/workload_processor.go:504]: The return type of `service.GetWaypoint().GetAddress()` is a struct, ref: https://github.com/kmesh-net/kmesh/blob/main/api/v2/workloadapi/workload.pb.go#L1292

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_processor.go:504]: my fault, misread service.Addresses as waypoint.Address

Comment (hzxuzhonghu) [reply pkg/controller/workload/workload_processor.go:504]: how could `service.Addresses` cause panic now

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**